gigascience/gigadb-website
# A11y - FAQ page

# Pull request for issue: #1381

This is a pull request for the following functionalities:

Fix accessibility issues in the FAQ page http://gigadb.gigasciencejournal.com:9170/site/faq

## How to test?

* Navigate to http://gigadb.gigasciencejournal.com:9170/site/faq and use an a11y audit tool (e.g. AXE chrome extension) to verify that there are no issues within the main content
* Navigate to http://gigadb.gigasciencejournal.com:9170/site/faq and check that the first element is expanded and the marker is "-"
* Navigate to http://gigadb.gigasciencejournal.com:9170/site/faq#panel02 and check that the second element is expanded and the marker is "-"

## How have functionalities been implemented?

* Fix heading hierarchy
* Add textual caption to complex image (last item)
* Replaced labels "here" of links with something more meaningful and less dependent on surrounding context
* Fixed initial state if url hash is not defined (the marker sign indicated that the first item was collapsed even though it was expanded, not it correctly indicates that it is expanded)
* Added aria attrbiutes and ids that connect each accordion content to its respective heading
* Used semantic html `<abbr>` element for abbreviations, and follow the standard of mentioning the abbreviation along the full text the first time, and the abbreviation plus a title attribute on subsequent times. Example:

```html
Data Publishing Charges (<abbr>DPCs</abbr>) (first mention) ... <abbr title="Data Publishing Charges">DPCs</abbr> (subsequent mentions)
```

* Use definition list for definition terms (`How do I report missing values in my metadata`)
* Updated page title to match current page

## Any issues with implementation?

* The accordions rely on bootstrap logic that uses `<a>` elements, so it's not trivial to replace by buttons which semantically would be more suitable. However they have `role="button"` which is not as good as using native buttons but acceptable
* http://www.gigasciencejournal.com/about/apcfaq/howmuch this link is referenced in the page but seems broken

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
